stages/stage_parsing_entry.go

```diff
@@ -53,22 +58,18 @@ func (s *StageParsingEntry) proceed() {
 		var entryMap common.EntryMap
 		var err error
 
-		for _, parse := range s.parsers {
-			entryMap, err = parse(message.Origin)
-
-			if err != nil {
-				s.logger.Warnf(
-					"Error parsing log entry, entry '%s', error '%s'",
-					message,
-					err,
-				)
-				continue
-			}
-
-			break
+		switch message.Format {
+		case common.CRITypeDocker:
+			entryMap, err = s.parseDockerFormat(message.Origin)
+		case common.CRITypeContainerD:
+			entryMap, err = s.parseContainerDFormat(message.Origin)
+		default:
+			err = ErrUnknownMessageFormat
```

Comment:
in case of a unknown format, it will be just parsed by parseDefault if I read this right. Should't you just make
```go
entryMap = s.parseDefault(message.Origin)
```
if you would like to log it, I would just add explicit log in default section as well

Reply:
"Default" parsing and error logging must be performed in case when format appears to be known (default case should not happen at all, because ProviderContainers doesn't produce containers with format other than these two), but log is malformed due to some reason.
So error should be checked anyway. So it seemed more clear to me to reduce "this-should-not-happen" case to error handling, not vice versa. What do you think?..